pkg/kubectl/cmd/create.go

```diff
@@ -75,19 +75,24 @@ func RunCreate(f *cmdutil.Factory, out io.Writer, filenames util.StringList) err
 		return err
 	}
 
-	cmdNamespace, err := f.DefaultNamespace()
+	cmdNamespace, overridden, err := f.DefaultNamespace()
 	if err != nil {
 		return err
 	}
 
 	mapper, typer := f.Object()
-	r := resource.NewBuilder(mapper, typer, f.ClientMapperForCommand()).
+	b := resource.NewBuilder(mapper, typer, f.ClientMapperForCommand()).
 		Schema(schema).
 		ContinueOnError().
-		NamespaceParam(cmdNamespace).RequireNamespace().
+		NamespaceParam(cmdNamespace).DefaultNamespace().
 		FilenameParam(filenames...).
-		Flatten().
-		Do()
+		Flatten()
+
+	if overridden {
```

Comment:
It would probably be cleaner to expose a method on the builder that takes both overridden and the namespace in one call.  

    .PreferNamespace(namespace string, mustMatch bool)